### PR TITLE
[LongT5] Remove duplicate encoder_attention_mask default value check

### PR DESCRIPTION
**Pull request by guillaume-be**
# What does this PR do?

This PR performs a minor code clean-up by removing duplicate code in the LongT5 stack. 
Currently, if the stack is a decoder block with `encoder_hidden_states` provided, the check for the existence of `encoder_attention_mask` is done twice:

1. https://github.com/huggingface/transformers/blob/c8f35a9ce37bd03f37fcf8336172bdcbe7ffc86a/src/transformers/models/longt5/modeling_longt5.py#L1452
2. https://github.com/huggingface/transformers/blob/c8f35a9ce37bd03f37fcf8336172bdcbe7ffc86a/src/transformers/models/longt5/modeling_longt5.py#L1479

Because the conditions on the second check are identical to that of the first check (`self.is_decoder is True`, `encoder_attention_mask is None`, `encoder_hidden_states is not None`), the second check will never be True since  `encoder_hidden_states` was updated during the second check.

This PR proposes removing the first check and only use the second when the extended encoder attention mask is set anyway.

## Who can review?
 @ArthurZucker , @younesbelkada

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks for fixing! The explanation looks clear to me
Can also confirm the slow tests pass

**Review by ArthurZucker — APPROVED** [review]
LGTM
